lock-api/src/main/java/com/palantir/lock/client/LockLeaseService.java

```diff
@@ -41,20 +41,23 @@
 import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
+import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 class LockLeaseService {
     private final NamespacedConjureTimelockService delegate;
     private final UUID clientId;
     private final LeaderTimeGetter leaderTimeGetter;
     private final BlockEnforcingLockService lockService;
+    private final Supplier<LockCleanupService> lockCleanupServiceSupplier;
```

Comment:
How important is is that there is only one for each LockLeaseService? I _think_ it's fine for factories to create their own view each time they need a LockCleanupService